website/content/docs/integrations/consul-integration.mdx

```diff
@@ -50,38 +50,207 @@ Consul data, Vault secrets, or just general configurations within a Nomad task.
-a Consul token to submit a job.
+The [Consul ACL][consul_acls] system protects the cluster from unauthorized
+access. When enabled, both Consul and Nomad must be properly configured in
+order for their integrations to work. Nomad agents must be configured with
+their own Consul ACL token, and Consul must be configured to accept workload
+identities from tasks and services running in Nomad.
 
-<Note>
+### Nomad Agents
 
-For Consul Enterprise users that would like to use Consul Namespaces, currently
-the only supported way of namespacing ACL objects is creating auth methods and
-binding rules in the `default` namespace, and roles and policies in the desired
-Consul namespace. The reason for this is simplicity: users wanting complete
-namespacing of all ACL objects would be required to create an auth method and
-binding rule per namespace. In the future we plan on adding such a possibility,
-but currently such a configuration is not supported by Nomad.
+Nomad agents need access to Consul in order to register themselves in the
+service catalog and discover other Nomad agents via service discovery for
+automatic clustering. Nomad servers also create [configuration
+entries][consul_config_entry] for Consul Service Mesh, so the specific
+permissions vary slightly between Nomad servers and clients.
 
-</Note>
+<Tabs>
+<Tab heading="Nomad Servers">
+
+```hcl
+agent_prefix "" {
+  policy = "read"
+}
+
+node_prefix "" {
+  policy = "read"
+}
+
+service_prefix "" {
+  policy = "write"
+}
+
+mesh = "write"
+```
+
+</Tab>
+<Tab heading="Nomad Clients">
+
+```hcl
+agent_prefix "" {
+  policy = "read"
+}
+
+node_prefix "" {
+  policy = "read"
+}
+
+service_prefix "" {
+  policy = "write"
+}
+```
+
+</Tab>
+</Tabs>
 
-### Configuring Consul Authentication
+### Nomad Workload Identities
 
-Create a configuration file for a Consul [JWT Auth Method][]. The `JWKSURL`
-field should point to all the Nomad servers; you should use a proxy or DNS A
-Record to ensure that requests can go to any server in the cluster. The
-remaining fields are required to match those shown here.
+Starting in Nomad 1.7, Nomad clients can use a task or service's [Workload
+Identity][nomad_wid] to authenticate to Consul and obtain an ACL token specific
+to the service or task. When using Nomad workload identities, you no longer
+need to pass in a Consul ACL token to submit a job.
+
+By default, Nomad does not generate workload identities for services, and tasks
+only have a default identity that is used to access Nomad itself. To access
+Consul, jobs must define additional workload identities suitable for Consul
+access.
+
+These additional identities can be defined as an [`identity`][] block inside
+`task` or `service` blocks. When specified inside a `task`, the
+[`identity.name`][] value must follow the pattern `consul_<cluster name>`,
+where `<cluster name>` is the value for [`consul.cluster`][] of the `consul`
```

Comment:
Similar to the vault example, it is not very clear to mention the `consul` block requiring a name and then it not being necessary and not shown on the code

Reply:
Ah good point, we actually _should_ add it to the example, even if not required, to nudge folks to start using (ref. https://github.com/hashicorp/nomad/issues/19390)

Reply:
Added `consul` block on CE example as well:
https://nomad-git-docs-consul-vault-integrations-hashicorp.vercel.app/nomad/docs/job-specification/identity#workload-identities-for-consul